### DISCUSSION TITLE
[QUESTION] webERP vs KwaMoja

### DISCUSSION THREAD

**Discussion by the-liquid-metal** [timschofield/webERP#206] [Development Discussion & Specification]
Hello.

Recently i found [KwaMoja](https://github.com/timschofield/KwaMoja).

1. Which one is the successor of the other?
2. What is the status of each project (still in development, discontinued)?
3. Are there any plans to use popular frameworks/library (laravel, symfony, Vue, React, etc)?
4. Are there any plans to use recent php version?

**Comment by timschofield** [comment]
Hello Hendra, I initially started KwaMoja at a time when my access to webERP was blocked. I now have got control of webERP back, and am busy porting all the KwaMoja improvements I did back to webERP. So the answer to questions 1 and 2 is that soon there will be no difference.
The answer to question 3 is no. The design philosophy of webERP remain unchanged. I believe that philosophy has been shown to work well for over 20 years. I realise many will disagree with this statement but that is what is great about Free/open source software.
As for 4, the master branch on this repository is mostly PHP8 compatible. There are just a few things I am working on before I can say categorically it is PHP8 compatible.

Hope this helps,
Tim

**Comment by the-liquid-metal** [comment]
> ... The design philosophy of webERP remain unchanged. ...
> 

I'm interested in this one. Can you explain it in more detail?

**Comment by timschofield** [comment]
Both Phil and I are accountants, with no background in coding, other than as a hobby. It was designed to be simple to understand, so that businesses could make their own changes, without a need for deep coding knowledge. Also the directory structure was intended to be very shallow. In general it was designed so that you can easily find your place in the code.
For example, if you are entering a GL journal you can see from the browser you are using GLJournal.php. Opening that file you can see all the logic involved in entering and processing a journal (there are a few includes, for session managemnt, header and footer).
We have always understood this breaks best practice but we have found over the last 20 years that businesses have been able to fairly easily modify the scripts for their own use.

Tim

**Comment by the-liquid-metal** [comment]
Thanks for the explanation.

I have known webERP for a long time. However, because my work has nothing to do with this kind of app, and webERP is a fairly complicated application, my knowledge of webERP is still very limited.

Hopefully i can contribute to this project.

**Comment by timschofield** [comment]
Your name rang a bell in connection with webERP. I think we had a brief exchange many years ago!

I am writing a to-do list of things outstanding on webERP. If it's OK with you I will let you know when the list is done and you can see if there are any items on the list you fancy working on.

Tim

**Reply by the-liquid-metal** [reply]
> Your name rang a bell in connection with webERP. I think we had a brief exchange many years ago!
> 
Oh really. I don't remember anything about this. I was involved in several open source projects. Some were just small contributions on bug reporting, feature requests, and other small things. I may have contributed to webERP.

> I am writing a to-do list of things outstanding on webERP. If it's OK with you I will let you know when the list is done and you can see if there are any items on the list you fancy working on.
>
Oh yes of course. I would be very happy to help.

**Comment by timschofield** [comment]
It's possibly another Hendra Gunawan. Looking at my records I had a brief exchange with a Hendra Gunawan in 2008 and in 2010. I have a very good memory for names!
Anyway, no matter I will let you know when the to-do list is done.

Tim

**Reply by the-liquid-metal** [reply]
Wow, that was a long time ago. I've never met anyone who remembers people they only had a brief communication with, and your  records must be very good.

At that time I was still in college. If I'm not mistaken, I found webERP and explored it at that time. Yeah, maybe that's me. But Hendra Gunawan is a common name in my country (Indonesia). Since then I haven't dug into webERP again. webERP came up a few times when I googled open source applications.
